### DISCUSSION TITLE
v0.11.1

### DISCUSSION THREAD

**Discussion by laipz8200** [langgenius/dify#10551] [Releases]
## ✨ What's New in v0.11.1? ✨

> [!Important]
>
> Heads up! Starting with the next minor version, we'll no longer support Python 3.10. Make sure to upgrade to Python 3.11 or 3.12 to stay current with our updates.

We're back with some fresh features and crucial fixes in version 0.11.1. Let’s dive in and see what's changed:

### 🚀 New Features

- **Vanna AI Base URL Configuration**: You can now configure the base URL for VannaAI, allowing more flexibility in routing requests by @BenjaminX in #10294.
  
- **Anthropic Claude-3-5-Haiku Support via OpenRouter**: Get ready to leverage the capabilities of Anthropic’s Claude-3-5-Haiku by @infinitnet in #10331.

- **GitLab Tools Integration**: Amp up your DevOps with the integration of GitLab tools by @wlrnet in #10407.

- **Video Understanding and CogVideo Tool Support**: Your LLM can now understand videos, and with the new CogVideo tool, process them too! Thanks to @hjlarry in #9828 and #10456.

- **Tool Search Enhancement**: You can now search for tools using the provider’s name, making it easier to find exactly what you need by @hjlarry in #10518.

- **Podcast Generator Tool Enhancements**: Set the OpenAI base URL when using our podcast generation tool, thanks to @XiaoLey in #10496 and #10517.

### ⚙️ Enhancements

- **SSRF Request Timeout Configuration**: New configuration options to customize SSRF request timeout for better security by @gubinjie in #10292.

- **PNG, GIF, and WebP Support**: More image formats for your media needs by @ZuzooVn in #7947.

- **Iteration Node Enhancements**: Parallel mode now maintains correct output order, and single run time per iteration is supported by @Nov1c444 in #10323 and #10512.

- **Celery Worker Log Format**: Logs now respect the `LOG_FORMAT` environment variable, making logs cleaner and more consistent by @liuhaoran1212 in #10016.

### 🛠️ Bug Fixes

- **ComfyUI Tool Prompt Fix**: Special prompts are back in action for the ComfyUI tool by @hjlarry in #10307.

- **Workflow and Node Fixes**: Various workflow-related bugs squashed, specifically handling optional values in workflow start nodes and ensuring no failures when variables go missing by @laipz8200 and @zxhlyh in #10471 and #10529.

---

### Docker compose deployments

1. Back up your customized docker-compose YAML file (optional)

   ```bash
   cd docker
   cp docker-compose.yaml docker-compose.yaml.$(date +%s).bak
   ```

2. Get the latest code from the main branch

   ```bash
   git checkout main
   git pull origin main
   ```

3. Stop the service，Command, please execute in the docker directory

   ```bash
   docker compose down
   ```

4. Back up data

   ```bash
   tar -cvf volumes-$(date +%s).tgz volumes
   ```

5. Upgrade services

   ```bash
   docker compose up -d
   ```

### Source Code deployments

1. Stop API server, Worker and Web frontend Server.

2. Get the latest code from the release branch:

   ```bash
   git checkout 0.11.1
   ```

3. Update Python dependencies:

   ```bash
   cd api
   poetry install
   ```

7. Then, let's run the migration script:

   ```bash
   poetry run flask db upgrade
   ```

8. Finally, run API server, Worker and Web frontend Server again.

---

## What's Changed
* fix: special prompt not work for comfyUI tool by @hjlarry in https://github.com/langgenius/dify/pull/10307
* docs: remove the TOC part by @laipz8200 in https://github.com/langgenius/dify/pull/10324
* feat(vannaai): add base_url configuration by @BenjaminX in https://github.com/langgenius/dify/pull/10294
* feat: add support for anthropic/claude-3-5-haiku through OpenRouter by @infinitnet in https://github.com/langgenius/dify/pull/10331
* feat: The SSRF request timeout configuration item is added by @gubinjie in https://github.com/langgenius/dify/pull/10292
* Gitee AI tools by @fchange in https://github.com/langgenius/dify/pull/10314
* Update README.md by @guchenhe in https://github.com/langgenius/dify/pull/10332
* feat: support png, gif, webp by @ZuzooVn in https://github.com/langgenius/dify/pull/7947
* fix(api): remove fixed source attribute from FileApi by @laipz8200 in https://github.com/langgenius/dify/pull/10353
* chore(ci): separate vector store tests into new workflow by @laipz8200 in https://github.com/langgenius/dify/pull/10354
* chore: update translation for 'account' from '계좌' to '계정' by @comfuture in https://github.com/langgenius/dify/pull/10350
* chore: lazy import sagemaker by @bowenliang123 in https://github.com/langgenius/dify/pull/10342
* chore(ci): bring back poetry cache to speed up CI jobs by @bowenliang123 in https://github.com/langgenius/dify/pull/10347
* fix: remove unsupported vision in OpenRouter Haiku 3.5 by @infinitnet in https://github.com/langgenius/dify/pull/10364
* fix(model_runtime): remove vision from features for Claude 3.5 Haiku by @mazyu36 in https://github.com/langgenius/dify/pull/10360
* fix: remove duplicated category “recommended” by @hjlarry in https://github.com/langgenius/dify/pull/10375
* fix typo: mMaximum -> Maximum by @y-omr in https://github.com/langgenius/dify/pull/10389
* Adjusted docker manifests and environment variables for OceanBase vector database by @powerfooI in https://github.com/langgenius/dify/pull/10395
* fixed: web api remote urls error by @luckylhb90 in https://github.com/langgenius/dify/pull/10383
* refactor(question_classifier): improve error handling with custom exceptions by @laipz8200 in https://github.com/langgenius/dify/pull/10365
* refactor(tool-node): introduce specific exceptions for tool node errors by @laipz8200 in https://github.com/langgenius/dify/pull/10357
* refactor(knowledge-retrieval): improve error handling with custom exceptions by @laipz8200 in https://github.com/langgenius/dify/pull/10385
* refactor(iteration): introduce specific exceptions for iteration errors by @laipz8200 in https://github.com/langgenius/dify/pull/10366
* fix(remote-files): fallback to get when remote server not support head method by @laipz8200 in https://github.com/langgenius/dify/pull/10370
* fix(conversation-service): return success response after conversation… by @BenjaminX in https://github.com/langgenius/dify/pull/10416
* chore: improve custom tool's display by @hjlarry in https://github.com/langgenius/dify/pull/10410
* feat(Tools): add lark tools by @hgnulb in https://github.com/langgenius/dify/pull/10117
* fix typo: Retrieve Chunks API Docs by @huangyafei in https://github.com/langgenius/dify/pull/10412
* chore: use posixpath to wrapper filepath by @yaoice in https://github.com/langgenius/dify/pull/9976
* chore(lint): Use logging.exception instead of logging.error by @bowenliang123 in https://github.com/langgenius/dify/pull/10415
* Conversation delete issue by @BenjaminX in https://github.com/langgenius/dify/pull/10423
* fix: simplify Enter key handling and remove unused ref by @SuzukiKatsuma in https://github.com/langgenius/dify/pull/10413
* fix: config violations when running db migtration ci tests by @bowenliang123 in https://github.com/langgenius/dify/pull/10428
* fix(http_request): send form data by @laipz8200 in https://github.com/langgenius/dify/pull/10431
* chore: use DEBUG in dify_config instead of parsing raw system environment variable in place by @bowenliang123 in https://github.com/langgenius/dify/pull/10437
* fix: (#10437 followup) fix conditions with DEBUG config by @bowenliang123 in https://github.com/langgenius/dify/pull/10438
* Feat/tools/gitlab by @wlrnet in https://github.com/langgenius/dify/pull/10407
* feat: support LLM understand video by @hjlarry in https://github.com/langgenius/dify/pull/9828
* fix(ops_tracing): enhance error handle in celery tasks. by @ZhouhaoJiang in https://github.com/langgenius/dify/pull/10401
* fix: correct output order in parallel mode for iteration nodes by @Nov1c444 in https://github.com/langgenius/dify/pull/10323
* chore: add MULTIMODAL_SEND_VIDEO_FORMAT to docker's env by @hjlarry in https://github.com/langgenius/dify/pull/10458
* feat: add cogVideo tool by @hjlarry in https://github.com/langgenius/dify/pull/10456
* Fix conversation response issue by @BenjaminX in https://github.com/langgenius/dify/pull/10450
* fix segment enable service api by @JohnJyong in https://github.com/langgenius/dify/pull/10445
* celery worker log format following LOG_FORMAT env#9404 by @liuhaoran1212 in https://github.com/langgenius/dify/pull/10016
* embedding model check when init the knowledge by @JohnJyong in https://github.com/langgenius/dify/pull/10463
* feat: Add support for complete domain names in the new URL prefix. by @QuietlyChan in https://github.com/langgenius/dify/pull/8893
* chore: make comfy workflow can generate image with a random seed by @hjlarry in https://github.com/langgenius/dify/pull/10462
* chore(ci): avoid reinstall pipx and pin poetry version aligned with in api dockerfile by @bowenliang123 in https://github.com/langgenius/dify/pull/10426
* update document and segment word count by @JohnJyong in https://github.com/langgenius/dify/pull/10449
* fix(migrations): correct schema reference in service API history migration by @laipz8200 in https://github.com/langgenius/dify/pull/10452
* refactor(core): Remove extra_config from File. by @laipz8200 in https://github.com/langgenius/dify/pull/10203
* add create tidb serverless job control by @JohnJyong in https://github.com/langgenius/dify/pull/10467
* feat(workflow-nodes): handle missing variables without failure by @laipz8200 in https://github.com/langgenius/dify/pull/10471
* fix(api): replace Raw field with FilesContainedField in MessageListApi inputs by @laipz8200 in https://github.com/langgenius/dify/pull/10472
* Fix/log tz by @crazywoola in https://github.com/langgenius/dify/pull/10473
* fix(file_upload): correct validation method and add unit tests by @laipz8200 in https://github.com/langgenius/dify/pull/10477
* fix: [VESSL-AI] edit some words in vessl_ai.yaml by @larcane97 in https://github.com/langgenius/dify/pull/10417
* Fix: Correct the max tokens of Claude-3.5-Sonnet-20241022 for Bedrock and VertexAI by @fdb02983rhy in https://github.com/langgenius/dify/pull/10508
* fix the error of unable to retrieve url from file by @XiaoLey in https://github.com/langgenius/dify/pull/10498
* feat(tools/podcast_generator): add support for setting openai base url with the podcast_generationor tool by @XiaoLey in https://github.com/langgenius/dify/pull/10496
* Revert "fix the error of unable to retrieve url from file" by @crazywoola in https://github.com/langgenius/dify/pull/10511
* feat(tools/podcast_generator): add support for setting openai base url with the podcast_generationor tool by @XiaoLey in https://github.com/langgenius/dify/pull/10517
* feat: support tool search also can search toolProvider's name by @hjlarry in https://github.com/langgenius/dify/pull/10518
* Support for incoming value modification by @charli117 in https://github.com/langgenius/dify/pull/10525
* The list action node adds methods to extract specific list objects by @charli117 in https://github.com/langgenius/dify/pull/10421
* refactor(code_executor): update input type annotations to use Mapping for better type safety by @laipz8200 in https://github.com/langgenius/dify/pull/10478
* add jina rerank http timout parameter by @JohnJyong in https://github.com/langgenius/dify/pull/10476
* Feat/iteration single run time by @Nov1c444 in https://github.com/langgenius/dify/pull/10512
* fix: workflow start node form optional value by @zxhlyh in https://github.com/langgenius/dify/pull/10529
* chore: (dockerfile) upgrade perl version by @Nov1c444 in https://github.com/langgenius/dify/pull/10534
* fix(model_runtime): ensure compatibility with O1 models by adjusting token parameters by @laipz8200 in https://github.com/langgenius/dify/pull/10537
* fix(gitee_ai): update English description for clarity and accuracy by @laipz8200 in https://github.com/langgenius/dify/pull/10540
* chore (vanna): update form parameter from 'form' to 'llm' in vanna.yaml by @BenjaminX in https://github.com/langgenius/dify/pull/10488
* fix(extractor): temporary file by @laipz8200 in https://github.com/langgenius/dify/pull/10543
* fix: iteration invalid output selector doesn't throw an error by @Nov1c444 in https://github.com/langgenius/dify/pull/10544
* chore: update version to 0.11.1 across all configurations and Docker images by @laipz8200 in https://github.com/langgenius/dify/pull/10539

## New Contributors
* @comfuture made their first contribution in https://github.com/langgenius/dify/pull/10350
* @huangyafei made their first contribution in https://github.com/langgenius/dify/pull/10412
* @SuzukiKatsuma made their first contribution in https://github.com/langgenius/dify/pull/10413
* @liuhaoran1212 made their first contribution in https://github.com/langgenius/dify/pull/10016
* @QuietlyChan made their first contribution in https://github.com/langgenius/dify/pull/8893
* @fdb02983rhy made their first contribution in https://github.com/langgenius/dify/pull/10508

**Full Changelog**: https://github.com/langgenius/dify/compare/0.11.0...0.11.1

<hr /><em>This discussion was created from the release <a href='https://github.com/langgenius/dify/releases/tag/0.11.1'>v0.11.1</a>.</em>

**Comment by XiaoLey** [comment]
@laipz8200 Uh... regarding the 'Bug Fixes' section, specifically the 'File URL Retrieval Issues', I actually resolved this issue, but not through this PR. This particular PR was a bit messy and got reverted. 😅 I made some adjustments, and it should actually be this PR: #10513.

**Reply by laipz8200** [reply]
Oh! I'll update it later. 🫣

**Comment by FBR65** [comment]
Hi, I just tried 0.11.1.

Uploading an image works now, but the name of the file on the server doesn't match the name or id which is given back.

![server](https://github.com/user-attachments/assets/d1e617c0-ff71-4d94-82e1-8be91c5df58a)
![upload_api](https://github.com/user-attachments/assets/c8cf68d8-e103-4071-bd47-89894242ea1d)

**Reply by laipz8200** [reply]
For safety reasons, we don't use file names entered by users in the storage path. Instead, the actual file names are stored in the database and retrieved when needed.

**Reply by FBR65** [reply]
Uploading file to Dify (Image/png) and then the ID given back from the API should be used to transfer the file to the backend.

So if the ID, which is given back not corresponding to the filename, what should be done?

**Reply by laipz8200** [reply]
You just need to use the ID, and Dify will be able to find the right file for you.

**Reply by FBR65** [reply]
No, that doesn't work. I tried it yesterday with 0.11.0 and today with 0.11.1.

My file is uploaded by /files/upload and afterwards the ID is given to Dify Client for sending the file and the query to Qwen VL.
But I'm alsways getting an error back.

It works  0.10.1 but not in 0.11.0 and 0.11.1.

**Comment by nadirvishun** [comment]
Is `0.11.1` version already support Python 3.11 or 3.12 ?

**Reply by bowenliang123** [reply]
Yes, it does. Tested on 3.10, 3.11 and 3.12.

**Comment by JiesenSun** [comment]
After upgrading to version 0.11.1 through docker-compose, the following information is printed crazily in the log.
![image](https://github.com/user-attachments/assets/f5448880-3b57-4033-b61d-69241aa9b3ae)

**Reply by fdb02983rhy** [reply]
Please have a check
https://github.com/langgenius/dify/issues/10555
https://github.com/langgenius/dify/pull/10563

**Reply by bowenliang123** [reply]
When and where have these logs are printed?
It seems you're running commands inside the worker service?
